rfcs/0095-empowering-new-scoped-module.md

```diff
@@ -0,0 +1,74 @@
+
+As in practices in other machine learning projects, there can be some levels of duplications or missing features compared with existing components (TorchFX TorchScript, TOSA and other MLIR graph IRs). Following the same practice in those related projects, as a team player in the community, one major principle in Apache is to empower communities by empowering optional components if they do not affect existing workflow. Empowering S0 through scoped modules brings a win-win situation for the community: it also brings in new aspirational members who are willing to collaborate and deliver the best for the community. This way, we keep ourselves up-to-date and grow stronger. On the other hand, failure to do so could result in community members getting discouraged and we lose valuable contributions and opportunities for us to grow in this rapidly growing area.
+
+The type of modules can include, but are not limited to:
+
+- IR dialects such as MLIR’s TOSA (while there are other graph IRs). TorchFX(while there is already TorchScript).
+- Vertical flows that leverage some of the dialects.
+- Backends/frontends.
+- Other types of modules introduced in a self-contained namespace.
+
+S0 changes would be contained in its namespace, with possible integrations also built inside its namespace. There can be follow-up steps (S1), such as making a dialect broadly accessible in existing compilation flows. Importantly, further S1/S2 level changes would require different RFC and longer deliberation for G0. The discussions on S1 also would serve as a way to allow the community to have a floor to talk about where broader areas of the project are going through a longer deliberation to maintain G0. Clearly identifying and empowering the S0 stage helps us to enable improvements quickly while bringing energies to the community, empowering a broader set of users, while not disrupting existing use cases.
+
+# Proposal: Empowering S0 Modules
+
+In this process RFC, We’d like to propose a process to encourage S0 modules and set expectations about what we anticipate in such inclusion.
+
+Note that this RFC focuses on the S0 stage. We propose the following guidelines to expedite to process while ensuring, quality and community support:
+
+- More than three PMC members endorse the S0-level proposal to ensure that there are enough eyes and shepherding in the module. The decision to establish a S0-level module needs to get majority support from PMC.
+- The code changes of S0-level modules follow the normal code review process as in all other modules in the codebase.*
+- A clear set of community members are committed to maintaining the proposed modules with technical support quantitatively, more than three endorsing committers who can serve as the initial owner.
```

Comment:
This actually seems overly restrictive in some regards. Requiring 3 endorsing committers who agree to own/maintain is a very high bar and would be exclusionary to newer contributors.

Reply:
We need a clear set of people who can review/maintain the code and maintain the overall review quality bar following the same process of all modules, so at least one is necessary. The volunteers also do not have to write code, but need to be responsible for code reviews and follow up actions, and contribute their time on a part-time basis. Likely the supporting PMC members would be able to serve that role initially. This being said, we anticipate new contributors being bought in as committers. The goal is to grow a set of contributions rather than restricting them.

Reply:
I suppose my question here is what would we lose by setting the threshold at a single owning committer? We already have a number of areas in the S1 code that aren't owned by 3 or more committers, so it's not clear what we'd gain from having a much higher requirement for S0 code. It's also quite a commitment to own code where you haven't been part of the development at all, so I think finding 3 volunteers could be troublesome for certain contributions.

Reply:
Thanks for the suggestion. The proposal makes sense to me. And I have updated the RFC from 3 committers to only one committer, which is aligned with typical code-changing PRs